### ISSUE TITLE
V0.8.0 support?

### ISSUE THREAD

**Issue by tanevanwifferen** [ipfs-shipyard/py-ipfs-http-client#252]
Can we just bump the required ipfs daemon version?

**Comment by ntninja** [comment]
I'm aiming to commit a huge set of changes before releasing 0.8 (again, its been delayed several times already). Please ping me in a week if you haven't heard anything from me before that.

**Comment by ntninja** [comment]
In general upgrading to a new go-IPFS works something like this (but do this yet):

  1. Update the minimum and maximum version in `ipfshttpclient/client/__init__.py` (next after 0.4.23 was 0.5.0)
  2. Update `.travis.yml` with matching go-ipfs versions for testing (both number and sha512 hash)
  3. Update `README.md` with matching version numbers in the note before the table of contents
  4. Open a PR
  5. Ensure that all checks (except for typing which is broken currently) pass, possibly requires patches and version workarounds to be written

^ Not all that bad, but it does take a bit of time.

**Comment by dokterbob** [comment]
Hey all, I'd like to strongly suggest to change `VersionMismatch` from an Exception into a Warning, allowing users to use whatever version they'd like. I would consider the current behaviour 'broken by design' from a user perspective; most IPFS behaviour doesn't change between versions but the current behaviour ensures that it still breaks and that seems unnecessary to me.

Of course having full official 0.8.0 support and other changes are super welcome. It's just IPFS 0.8 has been out for a while now and this is a major pain for users, which will occur again and again.

**Comment by ntninja** [comment]
I agree with this nowadays. Back in the 0.4.x days the breakage thing was pretty bad, but it's been a while and things seem to have settled down. Thank you for the suggestion!

**Comment by tanevanwifferen** [comment]
@ntninja any update? we're doing manual backups since the latest ipfs release...

**Comment by c0llab0rat0r** [comment]
@ntninja,

Also interested in a `0.8.0` release. Thanks.

**Comment by 0kalekale** [comment]
lads when are we getting v0.8.0 support

**Comment by c0llab0rat0r** [comment]
@ntninja has put a lot of time and effort into this great library.

With the interest and activity on it, I wonder if it needs an additional owner/shepherd.

**Comment by ntninja** [comment]
Somebody want to become co-maintainer? It appears I'm not as enthusiastic about maintaining this as I was when I took this over 5 years ago…

And 0.8.0 isn't the first release that has been greatly delayed for no strong reason either!

**Comment by ntninja** [comment]
After reviewing recent code I found minor issue #277 that should be fixed before that becomes part of the API in its current form. That's all that is required before actually releasing.

**Comment by c0llab0rat0r** [comment]
> Somebody want to become co-maintainer? It appears I'm not as enthusiastic about maintaining this as I was when I took this over 5 years ago…
> 
> And 0.8.0 isn't the first release that has been greatly delayed for no strong reason either!

I'm game to be a co-maintainer. It would be helpful to get any documentation gaps filled in related to release vetting (e.g. any manual steps taken above and beyond CI to ensure the library is ready to release) and related to the release/publish process.

There was also mention of some external commitments related to versions of Python etc. the library must support. It would be helpful to get more information on that.

**Comment by ntninja** [comment]
> I'm game to be a co-maintainer.

I was hoping you'd say (something like) that!

> It would be helpful to get any documentation gaps filled in related to release vetting

There a none, CI is fairly comprehensive as it is.

> related to the release/publish process

That is documented in https://github.com/ipfs-shipyard/py-ipfs-http-client/blob/master/docs/releasing.md . Wrote that a couple of years ago, but I kept updating it, so it should be fairly up-to-date.

> There was also mention of some external commitments related to versions of Python etc. the library must support. It would be helpful to get more information on that.

I've kinda established some rules on this over time, they currently go something like this:

  * The library must support for the last 5 go-IPFS versions (including patch versions) at the time of its release
  * The library must support all maintained CPython versions as provided by the core maintainers at python.org
  * The library must work on standard Python installations on Linux, macOS and also Windows
  * The library should support the latest PyPy version (which, until very recently, did not include CPython 3.6 compatibility)
  * The library should support all maintained CPython versions as found on Ubuntu Server standard maintenance versions (Ubuntu 16.04 just expired, oldest is now 18.04 with CPython 3.6 until 2023-04 – paid-support / extended security maintenance is not in scope)

I have not actually received any specific inquires about any of the above, but given the rather broad scope I don't believe anybody can expect more anyways. At the end of the day, this library *is* just a HTTP wrapper so I think its reasonable not dictate any overly strict version bounds for consumers. All the experimental stuff is happening within go-IPFS anyways.

I also wanted to extend the scope to js-IPFS at some point, but never have gotten around to that.

You have a PyPI account that I can add?

**Comment by c0llab0rat0r** [comment]
@ntninja I've created a PyPi account with the same name, c0llab0rat0r.

**Comment by ntninja** [comment]
@c0llab0rat0r: Added you on PyPI.

**Comment by ntninja** [comment]
Wanna try doing the 0.8 release?

**Comment by rounin-rp** [comment]
Hello guys. I am a newbie in ipfs so I don't understand much. Just need to ask when can we expect 0.8 support. I have a project that I need to submit on 22nd May.
Or if you could tell me how to run v0.7 that would be life saving help
Thanks

**Comment by c0llab0rat0r** [comment]
> Wanna try doing the 0.8 release?

See #280

**Comment by c0llab0rat0r** [comment]
@dokterbob do you have any interest in helping with moderation/admin of this library?

cc @ntninja

**Comment by dokterbob** [comment]
@c0llab0rat0r Not really, I don't have the time for it, but I am very happy and willing to make a 0.7.1 patch release so we finally can get this d#&$# thing to run! I've been waiting many months now just to get my package out.

**Comment by dokterbob** [comment]
That is to say, I might be willing to support release engineering on a one-off basis. In addition, I'd be willing to create a `RELENG.md` document with clear release instructions.

**Comment by c0llab0rat0r** [comment]
Version `0.8.0a1` has been published to PyPi. Version `0.8.0a2` will include some documentation fixes but no functional changes.

**Comment by Luflosi** [comment]
Can you please also tag the releases for GitHub?
